service/lib/agama/proxy_setup.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2023] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "uri"
+
+module Agama
+  # This class is responsible of parsing the proxy url from the kernel cmdline or configured
+  # during the boot proccess of the system writing the configuration to /etc/sysconfig/proxy
```

Comment:
It would be nice to mention that the class tries to read the proxy settings from dracut and the kernel's command line.

Reply:
That is what the 'or configured during the boot proccess..' referred, the thing is that it is still not implemented.